### PR TITLE
refactor: modify manifests for PY templates, add CEAEnabled consume logic and add test case

### PR DESCRIPTION
**Pull request by frankqianms**
workitem: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/29830998

### CODE REVIEW

**Review by yukun-dong — DISMISSED** [review]

File: templates/python/custom-copilot-basic/appPackage/manifest.json.tpl [hunk 1]
```diff
@@ -1,7 +1,14 @@
 {
+    {{#CEAEnabled}}
+    "$schema": "https://raw.githubusercontent.com/OfficeDev/microsoft-teams-app-schema/preview/DevPreview/MicrosoftTeams.schema.json",
```
Comment (blackchoey) [inline templates/python/custom-copilot-basic/appPackage/manifest.json.tpl:3]: The url should be https://developer.microsoft.com/en-us/json-schemas/teams/vdevPreview/MicrosoftTeams.schema.json (thought the CEA node is not defined here..)

File: templates/python/custom-copilot-rag-azure-ai-search/appPackage/manifest.json.tpl [hunk 2]
```diff
@@ -1,7 +1,14 @@
 {
+    {{#CEAEnabled}}
+    "$schema": "https://raw.githubusercontent.com/OfficeDev/microsoft-teams-app-schema/preview/DevPreview/MicrosoftTeams.schema.json",
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-azure-ai-search/appPackage/manifest.json.tpl:3]: also update the schema link here

File: templates/python/custom-copilot-rag-custom-api/appPackage/manifest.json.tpl [hunk 3]
```diff
@@ -1,7 +1,14 @@
 {
+    {{#CEAEnabled}}
+    "$schema": "https://raw.githubusercontent.com/OfficeDev/microsoft-teams-app-schema/preview/DevPreview/MicrosoftTeams.schema.json",
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-custom-api/appPackage/manifest.json.tpl:3]: also update the schema link here

File: templates/python/custom-copilot-rag-custom-api/appPackage/manifest.json.tpl [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
+    "manifestVersion": "devPreview",
+    "version": "1.0.1",
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-custom-api/appPackage/manifest.json.tpl:5]: do we need to change the version to `1.0.1`? If not, let's change all the manifest's version to 1.0.0

**Review by blackchoey — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

**Pull request merged**